File: Habitica/src/main/java/com/habitrpg/android/habitica/ErrorResponse.java
package com.habitrpg.android.habitica;

import com.magicmicky.habitrpgwrapper.lib.models.HabiticaError;

import java.util.List;

public class ErrorResponse {
    public String message;
    public List<HabiticaError> errors;

    public String getDisplayMessage() {
        if (errors != null && errors.size() > 0) {
            HabiticaError error = errors.get(0);
            if (error.message != null && error.message.length() > 0) {
                return error.message;
            }
        }
        if (message != null && message.length() > 0) {
            return message;
        }
        return "";
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/responses/ErrorResponse.kt
package com.habitrpg.android.habitica.models.responses

import com.habitrpg.common.habitica.models.responses.HabiticaError

class ErrorResponse {
    var message: String? = null
    var errors: List<HabiticaError>? = null
    val displayMessage: String
        get() {
            if (errors?.isNotEmpty() == true) {
                val error = errors?.get(0)
                if (error?.message?.isNotBlank() == true) {
                    return error.message ?: ""
                }
            }
            return message ?: ""
        }
}
